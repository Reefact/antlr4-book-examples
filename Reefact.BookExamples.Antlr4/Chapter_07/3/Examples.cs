#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_07._3 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        public void mermaid_tree_style_of_t_properties() {
            // Setup
            GRun grun = GRun.ReadResource("t.properties", 7, 3);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void load_t_properties() {
            // Setup
            GRun grun = GRun.ReadResource("t.properties", 7, 3);
            // Exercise
            string properties = grun.ToPropertiesString();
            // Verify
            Approvals.Verify(properties);
        }

    }

}