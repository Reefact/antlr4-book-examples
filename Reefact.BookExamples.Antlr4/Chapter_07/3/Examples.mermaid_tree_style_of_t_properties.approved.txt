graph TD
	1["file"] --> 2["prop"]
	2 --> 3["user"]
	2 --> 4["="]
	2 --> 5["#0034;parrt#0034;"]
	2 --> 6["#0092;r#0092;n"]
	1 --> 7["prop"]
	7 --> 8["machine"]
	7 --> 9["="]
	7 --> 10["#0034;maniac#0034;"]
	7 --> 11["#0092;r#0092;n"]
